general/cuda.hpp

```diff
@@ -15,17 +15,15 @@
 #include "../config/config.hpp"
 #include "error.hpp"
 
-#ifdef MFEM_USE_CUDA
-#include <cuda_runtime.h>
-#include <cuda.h>
-#endif
-
 // CUDA block size used by MFEM.
 #define MFEM_CUDA_BLOCKS 256
 
 #ifdef MFEM_USE_CUDA
 #define MFEM_DEVICE __device__
+#define MFEM_LAMBDA __host__
 #define MFEM_HOST_DEVICE __host__ __device__
+#define MFEM_DEVICE_SYNC MFEM_GPU_CHECK(cudaDeviceSynchronize())
+#define MFEM_STREAM_SYNC MFEM_GPU_CHECK(cudaStreamSynchronize(0))
```

Comment:
Can you add some documentation to give the intuition when one should be used over the other?

Reply:
`done`